front/migrations/20250408_missing_conversation_participants.ts

```diff
@@ -0,0 +1,168 @@
+import _ from "lodash";
+import { Op, UniqueConstraintError } from "sequelize";
+
+import {
+  ConversationModel,
+  ConversationParticipantModel,
+  Message,
+  UserMessage,
+} from "@app/lib/models/assistant/conversation";
+import { frontSequelize } from "@app/lib/resources/storage";
+import logger from "@app/logger/logger";
+import { makeScript } from "@app/scripts/helpers";
+
+const CHUNK_SIZE = 1000;
+
+/**
+ * Find all conversations in the workspace that have missing participants and process them.
+ * @param workspaceId - The workspace model ID to process. Not the sId.
+ * @param execute - Whether to execute the migration.
+ */
+async function processConversationsWithMissingParticipants(
+  workspaceId: number,
+  execute: boolean
+) {
+  // Find all conversations in the workspace that have missing participants.
+  const conversationsWithMissingParticipants = await ConversationModel.findAll({
+    where: {
+      workspaceId,
+      id: {
+        [Op.notIn]: frontSequelize.literal(
+          `(SELECT DISTINCT "conversationId" FROM conversation_participants)`
```

Comment:
maybe add workspaceId to avoid a too huge list of conversationIds ?

Reply:
Definitely!